internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -325,10 +328,37 @@ func dataSourceName(connDetails *ConnectionDetails) string {
 		// NOTE: as things stand the resulting DSN here means that if TLS is available
 		// it'll be used between Secretless and the server.
 		// To disable TLS altogether we must change to
-		// "sqlserver://%s:%s@%s?encrypt=disable",
-		"sqlserver://%s:%s@%s",
+		// "sqlserver://%s:%s@%s?encrypt=%s",
+		"sqlserver://%s:%s@%s?encrypt=%s",
```

Comment:
Hmm, out of scope but... the above raises the question of when this error should come to the attention of the user. I actually think this should happen when the Secretless config is loaded because this is a config time error. This should probably be the behaviour across the board the board with all the connectors

Reply:
So this is a conversation we've had a few times, about whether or not we should be validate these specific params. If we introduce this here, it will require validation for each of the other params, documentation on the errors, and so on. Some people had strong opinions on this topic, so I agree that this may be a conversation for another time / another story.

Reply:
I'm fine either way, I think we should create a story for it and lay out how we want to do it.
My only two thoughts are that we should:
  - Make stories for other adapters having the same validation, for consistency of user experience
  - Make sure that our error handling doesn't disrupt the user workflow or obfuscate important error information from the server

Reply:
@BradleyBoutcher  Agreed, let's create issues.